### PR TITLE
Change std::string parameters in cudf::strings APIs to std::string_view

### PR DESCRIPTION
**Pull request by davidwendt**
Follow on to #10810. This changes other occurrences of parameter type `std::string` to `std::string_view` in the non-regex APIs. 
This covers `cudf::strings::pad` fill-char and the to/from-timestamp/duration converters that accept format specifiers.

### CODE REVIEW

File: cpp/src/strings/convert/convert_durations.cu [hunk 1]
```diff
@@ -84,10 +84,10 @@ struct alignas(4) format_item {
  * components and when formatting a string from duration components.
  */
 struct format_compiler {
-  std::string format;
+  std::string_view const format;
   rmm::device_uvector<format_item> d_items;
-  format_compiler(const char* format_, rmm::cuda_stream_view stream)
-    : format(format_), d_items(0, stream)
+  format_compiler(std::string_view format, rmm::cuda_stream_view stream)
+    : format(format), d_items(0, stream)
```
Comment (hyperbolic2346) [inline cpp/src/strings/convert/convert_durations.cu:90]: Should this be renamed to avoid the `format(format)` assignment?

**Review by hyperbolic2346 — CHANGES_REQUESTED** [review]
Looks good

File: cpp/src/strings/convert/convert_durations.cu (hunk 1 above)
Comment (davidwendt) [reply cpp/src/strings/convert/convert_durations.cu:90]: No, the compiler handles this fine and is supported by the C++ standard (section 12.6.2 I believe).
Using the same name where the argument directly sets a member variable in the ctor initializer reinforces they are literally the same thing.

Comment (hyperbolic2346) [reply cpp/src/strings/convert/convert_durations.cu:90]: I know it is technically valid, but I thought the guide was to use the `format_` nomenclature. I'm not strongly opinionated here though if you'd like to keep it.

Comment (hyperbolic2346) [reply cpp/src/strings/convert/convert_durations.cu:90]: I absolutely would not call it necessary.

**Review by hyperbolic2346 — APPROVED** [review]

**Comment by davidwendt** [comment]
@gpucibot merge

**Pull request merged**